### DISCUSSION TITLE
Building list including links not working

### DISCUSSION THREAD

**Discussion by gasperphoenix** [SilentVoid13/Templater#1238] [Help]
Hi, in my frontmatter I have a attribute _author_ including 1..n authors and I would like to generate a comma separated list of links to the author pages from the attribute. What I did so far:

Frontmatter of the file
```
---
author:
   - Kevin Leman
   - William Pentak
---
```

In the note itself I would like to generate a list of author links from this attribute. Therefore I wrote a custom script "list_to_link_list.js":

```JavaScript
function my_function(comma_separated_list) {
    var list, link_list = "", i;
    
    list = String(comma_separated_list).split(",");
    
    for (i = 0; i < list.length; i++)
    {
        link_list += "[[" + String(list[i]) + "]]";
        
        if (i != list.length-1)
        {
            link_list += ", ";
        }
    }
        
    return (link_list);
}
module.exports = my_function;
```

I invoke the script using the following command:

```
<%*+ tR += tp.user.list_to_link_list(tp.frontmatter["author"]) %>
```

It generates the list but it does not interpret the names as links to the author pages.

```
Authors: [[Kevin Leman]], [[William Pentak]]
```

I'm new to JavaScript and Templater. Do I have to do something additional to achieve this?

Thanks

**Comment by Zachatoo** [comment]
If I had to guess, it's because you're using a [dynamic command](https://silentvoid13.github.io/Templater/commands/dynamic-command.html). I'd recommend not using those in general, they tend to be buggy and are planned to be deprecated (#913).

If you're okay with the content being static once you execute the template, you can remove the `+` to make it not dynamic.

```js
<%* tR += tp.user.list_to_link_list(tp.frontmatter["author"]) %>
```

If you need it to be dynamic, I highly recommend the Dataview plugin for dynamic content.
